packages/react/src/components/MenuButton/index.tsx

```diff
@@ -115,10 +117,48 @@ const MenuButton = forwardRef<HTMLDivElement, MenuButtonProps>(
     const prefix = usePrefix();
     const triggerRef = useRef<HTMLDivElement>(null);
     // eslint-disable-next-line @typescript-eslint/no-explicit-any -- https://github.com/carbon-design-system/carbon/issues/20452
-    let middlewares: any[] = [];
+    const middlewares: any[] = [];
```

Comment:
This entire middleware setup needs to be refactored. Instead of an empty array and imperatively `push()`ing options, it should be more declarative like all the other usages of floating-ui.

```diff
- middlewares.push(middleware1());
- if (aCondition) {
- 
-  // what if this actually needs to come before middleware1, for instance?
-  middlewares.push(middleware2()); 
- 
- }

+ middlewares: [
+  middleware1(),
+  aCondition && middleware2(),
+ ]
```

This way we're not running the risk of something being `push` ed too late impacting the order of the middleware's, which does matter.